src/dbnode/network/server/tchannelthrift/node/service.go

```diff
@@ -124,6 +132,8 @@ type service struct {
 	sync.RWMutex
 
 	db      storage.Database
+	dbIsSet uint64 // Atomic, 1 for true 0 for false.
```

Comment:
Can you just use the existing mutex instead of using an atomic?  Makes the code much more idiomatic.